# Computational Scoring and Experimental Evaluation of Enzymes Generated by Neural Networks



## Colab notebooks

| Notebook | Description |
| -------- | ----------- |
| [ESM-MSA sampler](https://colab.research.google.com/github/seanrjohnson/protein_scoring/blob/main/colab_notebooks/ESM_MSA_sequence_generation.ipynb) | uses the ESM-MSA model (a transformer-based neural network trained on protein multiple sequence alignments) to generate new protein sequences by iteratively mutating sequences from an input alignment. |
| [Metrics](https://github.com/seanrjohnson/protein_scoring/blob/main/colab_notebooks/Protein_Metrics.ipynb) | Calculates various sequence- and structure-based quality scores for proteins, such as those produced by generative models. | 


## Figures

### Setup

```shell
conda env create --name protein_scoring -f conda_env.yml

jupyter lab
```

## Related data and repositories

- [Source data](https://zenodo.org/record/7808981): AlphaFold2 predicted structures, Full sequence lists, Tables of metrics, Tables of experimental results, Phylogenetic Trees. Jupyter notebooks under "notebooks_for_figures" will automatically download the necessary data from Zenodo, but if you want it for some other purpose, it's available at this link.
- [protein_gibbs_sampler](https://github.com/seanrjohnson/protein_gibbs_sampler): command line tools for generating new sequences using ESM-MSA sampling (used in the notebook above). 

## References

- Johnson, Sean R., Xiaozhi Fu, Sandra Viknander, Clara Goldin, Sarah Monaco, Aleksej Zelezniak, and Kevin K. Yang. “Computational Scoring and Experimental Evaluation of Enzymes Generated by Neural Networks.” bioRxiv, March 4, 2023. [https://doi.org/10.1101/2023.03.04.531015](https://doi.org/10.1101/2023.03.04.531015).


